### DISCUSSION TITLE
Eturnal memory usage

### DISCUSSION THREAD

**Discussion by notEvil** [nextcloud/all-in-one#3365] [Questions] [answered]
Hi,

eturnal is using 1GB of memory which is a lot imo. Is this intended?

**Comment by szaimen** [comment]
No, this is not intended. On my test instance eturnal only takes 20MB RAM. 

How many devices (talk-android and talk-ios) are using your server and did you try if a restart of your server fixes this?

**Reply by notEvil** [reply]
After restart and watching `glances` the process starts with 2GB res (5GB virt) and after a few minutes drops to 800MB res. It then seems to release memory very slowly in certain intervals. The instance is mostly single user (myself) and talk is not used at all. The logs are clean btw and the instance is otherwise fully operational.

**Reply by notEvil** [reply, marked as answer]
sry for the distraction, it was my fault! (unknowingly messed with ulimit and eturnal seems to take what it gets)

**Comment by Vatson112** [comment]
If someone be here - here what you need to put in compose file of your service.

```yaml
 ulimits:
      nofile:
        soft: 65536
        hard: 65536
      nproc: 1024
```

By adding this settings RSS of process change from 2GB to 80MB.
